### PR TITLE
Fixing a bug the causes FaceOrigin collections to face the wrong direction when collections are rotated (and some bonuses) TAKE 2

### PR DESCRIPTION
**Pull request by Ecnassianer**
I dug myself into a githole in that other branch. I gave up trying to revert my reverts of merges of merges and just put my changes in a new branch. Sorry for making you check it twice. It should have all the changes I made in response to previous review feedback. The only bit we should be waiting for is @paseb to weigh in on the RadialMapping() question. I'll poke him in person tomorrow.

image

---------------------------------------------
This is a pull request for the completion of my proposal:
#2227

### CODE REVIEW

**Comment by paseb** [comment]
@StephenHodgson, I couldn't add directly to the comments but here my responses to the above comments.  Radial mapping represents a radial menu so the grid source is not what is being used though there's no really need to make a new vector so assigning it directly is fine.  Since this is a single relative axis rotation gimbal lock isn't an issue.  If we were going to do any multi-axis rotations we should be using matrix transforms.

**Comment by keveleigh** [comment]
Cool, so it looks like addressing the "update source vs new vector creation" concern is all that's needed here for approval.

File: Assets/HoloToolkit/UX/Scripts/Collections/ObjectCollection.cs [hunk 1]
```diff
@@ -62,23 +62,94 @@ public class ObjectCollection : MonoBehaviour
...
+
+        public float CellHeight
+        {
+            get { return cellHeight; }
+            set { cellHeight = value; }
+        }
+
+        /// <summary>
+        /// Margin between objects horizontally.
+        /// </summary>
```
Comment (StephenHodgson) [inline Assets/HoloToolkit/UX/Scripts/Collections/ObjectCollection.cs]: Summary tags should be above the public properties not the attributes for the private fields

File: Assets/HoloToolkit/UX/Scripts/Collections/ObjectCollection.cs [hunk 2]
```diff
@@ -395,6 +396,65 @@ public void UpdateCollection()
...
+
+                case OrientTypeEnum.FaceOriginReversed:
+                    node.transform.rotation = Quaternion.LookRotation(this.transform.position - node.transform.position, this.transform.up);
+                    break;
+
+                case OrientTypeEnum.FaceCenterAxis:
+                    centerAxis = Vector3.Project(node.transform.position - this.transform.position, this.transform.up);
+                    pointOnAxisNearestNode = this.transform.position + centerAxis;
+                    node.transform.rotation = Quaternion.LookRotation(node.transform.position - pointOnAxisNearestNode, this.transform.up);
+
```
Comment (StephenHodgson) [inline Assets/HoloToolkit/UX/Scripts/Collections/ObjectCollection.cs]: a couple of extra lines before breaks.

File: Assets/HoloToolkit/UX/Scripts/Collections/ObjectCollection.cs [hunk 3]
```diff
@@ -434,6 +494,27 @@ private Vector3 CylindricalMapping(Vector3 source, float radius)
...
+        /// <param name="source">The source <see cref="Vector3"/> to be mapped to cylinder</param>
+        /// <param name="radius">This is a <see cref="float"/> for the radius of the radial</param>
+        /// <param name="row">This is a <see cref="int"/> for the radius of the radial</param>
+        /// <param name="column">This is a <see cref="int"/> for the radius of the radial</param>
+        /// <returns></returns>
+        private Vector3 RadialMapping(Vector3 source, float radius, int row, int column)
+        {
+            Radius = radius >= 0 ? Radius : radius;
+       
+            Vector3 newPos = new Vector3(0f, 0f, (Radius/Rows) * row);
```
Comment (StephenHodgson) [inline Assets/HoloToolkit/UX/Scripts/Collections/ObjectCollection.cs]: can we just assign the vector instead of creating a new one?

**Pull request merged**